geoportal/geoportailv3_geoportal/lib/lingua_extractor.py

```diff
@@ -0,0 +1,378 @@
+            elements = re.split(r'(?<!\\),', attributes_to_remove)
+        elements.extend([geometry_column, 'st_asgeojson'])
+        for element in elements:
+            try:
+                del attributes[element.replace("\\,", ",")]
+            except:
+                pass
+        return attributes
+
+    def _ogc_getfeatureinfo(self, session, url, x, y, width, height,
+                            layer, bbox, srs, layer_id):
+        # session = get_session('development.ini', 'app')
+        from c2cgeoportal_commons.models.main import Metadata
+
+        body = {
+            'SERVICE': 'WMS',
+            'VERSION': '1.1.1',
+            'REQUEST': 'GetFeatureInfo',
+            'QUERY_LAYERS': layer,
+            'LAYERS': layer,
+            'STYLES': '',
+            'INFO_FORMAT': 'application/json',
+            'FEATURE_COUNT': '50',
+            'X': x,
+            'Y': y,
+            'SRS': srs,
+            'WIDTH': width,
+            'HEIGHT': height,
+            'BBOX': bbox
+        }
+        metadata = session.query(Metadata).filter(Metadata.item_id == layer_id).\
+                   filter(Metadata.name == "ogc_layers").first()
+        if metadata is not None:
+            body['LAYERS'] = metadata.value
+
+        metadata = session.query(Metadata).filter(Metadata.item_id == layer_id).\
+                   filter(Metadata.name == "ogc_query_layers").first()
+        if metadata is not None:
+            body['QUERY_LAYERS'] = metadata.value
+
+        metadata = session.query(Metadata).filter(Metadata.item_id == layer_id).\
+                   filter(Metadata.name == "ogc_info_format").first()
+        if metadata is not None:
+            body['INFO_FORMAT'] = metadata.value
+        ogc_info_srs = "epsg:2169"
+        metadata = session.query(Metadata).filter(Metadata.item_id == layer_id).\
+                   filter(Metadata.name == "ogc_info_srs").first()
+        if metadata is not None:
+            ogc_info_srs = metadata.value
+
+        separator = "?"
+        if url.find(separator) > 0:
+            separator = "&"
+        query = '%s%s%s' % (url, separator, urlencode(body))
+        content = ""
+        try:
+            result = urllib.request.urlopen(query, None, 15)
+            content = result.read()
+        except Exception as e:
+            log.exception(e)
+            return []
+        try:
+            ogc_features = geojson_loads(content)
+            return (dict((key, key)
+                         for key, value in ogc_features['features'][0]['properties'].items()))
+        except Exception as e:
+            log.exception(e)
+            return []
+        return []
+
+    def _extract_messages(self, session):  # pragma: nocover
+        print('Entering into the tooltip extractor')
+
+        from c2cgeoportal_commons.models import DBSession, DBSessions
+        from geoportailv3_geoportal.models import LuxGetfeatureDefinition, LuxLayerInternalWMS
```

Comment:
Same, Database session seams like an arg of the function.

Reply:
done